source/code/include/playfab/PlayFabWinHttpPlugin.h

```diff
@@ -36,9 +36,13 @@ namespace PlayFab
         void HandleResults(std::unique_ptr<CallRequestContainer> requestContainer);
         void SetErrorInfo(CallRequestContainer& requestContainer, const std::string& errorMessage, const int httpCode = 0) const;
 
+        BOOL TryAddHeader(HINTERNET hRequest, LPCWSTR lpszHeaders);
+        void CompleteRequest(std::unique_ptr<CallRequestContainer> requestContainer, HINTERNET hRequest, HINTERNET hConnect, HINTERNET hSession);
+
         std::thread workerThread;
         std::mutex httpRequestMutex;
         std::atomic<bool> threadRunning;
+        std::atomic<BOOL> setPredefinedHeadersFailed;
```

Comment:
>BOOL [](start = 20, length = 4)

I still think a better choice for this value is std::atomic<HRESULT>

if you do that then TryAddHeader can return HRESULTs and can locally do the conversion from WinHttp's bool return to a more synchronous HRESULT and I think that will noticeably improve readability

Here are some examples of where the code changes and improves IMO
```
// ...
// Add HTTP headers
SetPredefinedHeaders(reqContainer, hRequest);
HRESULT hr = setPredefinedHeadersResult.exchange(S_OK);
if(FAILED(hr))
{
    SetErrorInfo(reqContainer, "Error in attempting to add Default Headers with HRESULT: " + std::to_string(hr));
    CompleteRequest(std::move(requestContainer), hRequest, hConnect, hSession);
    return;
}
// ..

// ..
std::string header = obj.first + ": " + obj.second;
HRESULT hr = TryAddHeader(hRequest, std::wstring(header.begin(), header.end()).c_str());
if(FAILED(hr))
{
    SetErrorInfo(reqContainer, "Error in WinHttpAddRequestHeaders attempting to add parameters with HRESULT: " + std::to_string(hr));
    CompleteRequest(std::move(requestContainer), hRequest, hConnect, hSession);
    return;
}
// ...

void PlayFabWinHttpPlugin::SetPredefinedHeaders(const CallRequestContainer& requestContainer, HINTERNET hRequest)
{
    UNREFERENCED_PARAMETER(requestContainer);
    
    HRESULT hr = S_OK;

    hr = TryAddHeader(hRequest, L"Accept: application/json");
    if (SUCCEEDED(hr))
    {
        hr = TryAddHeader(hRequest, L"Content-Type: application/json; charset=utf-8");
        if (SUCCEEDED(hr))
        {
            std::string versionHeader = "X-PlayFabSDK: " + PlayFabSettings::versionString();
            std::wstring versionWideHeader(versionHeader.begin(), versionHeader.end());
            hr = TryAddHeader(hRequest, versionWideHeader.c_str());
            if (SUCCEEDED(hr))
            {
                hr = TryAddHeader(hRequest, L"X-ReportErrorAsSuccess: true");
            }
        }
    }
    
    setPredefinedHeadersResult = hr;
}

HRESULT PlayFabWinHttpPlugin::TryAddHeader(HINTERNET hRequest, LPCWSTR lpszHeaders)
{
    bool succeeded = WinHttpAddRequestHeaders(hRequest, lpszHeaders, -1, 0);
    if (!succeeded)
    {
        return HRESULT_FROM_WIN32(GetLastError());
    }
    return S_OK;
}
```